doc/source/whatsnew/v0.24.1.rst

```diff
@@ -63,6 +63,9 @@ Bug Fixes
 -
 -
 
+**Reshaping**
+
+- Bug in :func:`merge` when merging by index name would sometimes result in an incorrectly numbered index (:issue:`24212`)
```

Comment:
I'm not sure if we are creating additional sections in the `whatsnew`...

cc @TomAugspurger

Reply:
There's a reshaping section in the 0.24.0 one. If not under reshaping, would it be correct to put this under indexing?